### ISSUE TITLE
[Feature Request] accessibility of the to_top_button

### ISSUE THREAD

**Issue by neralex** [bootscore/bootscore#1060]
### Describe the feature that you'd like

While fixing some accessibility issues based on the WCAG 2.1 rules, I just stumbled across this pseudo-button. From a technical perspective, it doesn't really make sense to use an anchor-tag, since this element mimics the functionality of a button, but it doesn't have an associated role-attribute, which wouldn't be necessary if the element is a vanilla HTML button tag. And if aria-label is already added to the element, then the nested span.visually-hidden-focusable is no longer necessary because aria-label already does the job.

So I had to override the footer.php from the main-theme to the child theme to change it to:

`<button class="<?= apply_filters('bootscore/class/footer/to_top_button', 'btn btn-primary shadow'); ?> position-fixed zi-1000 top-button" aria-label="<?php esc_attr_e('Return to top', 'bootscore' ); ?>"><?= apply_filters('bootscore/icon/chevron-up', '<i class="fa-solid fa-chevron-up" aria-hidden="true"></i>'); ?></button>`

But then the scroll-functionality doesn't work anymore and I had to add a extra click-function for this simple button. 

1. Where is the scroll-functionality in the main-theme placed for this pseudo-button?
2. Is there chance to change this pseudo-button to an vanilla HTML button without the nested span.visually-hidden-focusable to avoid override the whole footer.php from the main theme?

Thanks in advance

### Checklist

- [x] I have searched the issues and this feature has not been requested.

**Comment by crftwrk** [comment]
> Where is the scroll-functionality in the main-theme placed for this pseudo-button?

There is no scroll-functionality in the theme , the to-top button uses a simple `href="#"` which scrolls to the top of the page and scrolling effect is made by default Bootstrap:

```css
html {
  scroll-behavior: smooth;
}
```


This is a simple and modern way to add a to-top button as written by [CSS-Tricks](https://css-tricks.com/how-to-make-an-unobtrusive-scroll-to-top-button/#aa-and-what-about-smooth-scrolling).

In version 4, we had a button with JS but luckily since all modern browsers supports `scroll-behavior` there is no need for this anymore.

If you want to change the `<a>` to a `<button>` element, you have to add additional JS.

The only JS still exists is to hide/show the button on scroll position: 

https://github.com/bootscore/bootscore/blob/a0950beef48bd3bf2b1413519a6bd2ddba41720b/assets/js/theme.js#L23-L32


> Is there chance to change this pseudo-button to an vanilla HTML button without the nested span.visually-hidden-focusable to avoid override the whole footer.php from the main theme?

The button is coded in the `footer.php`. So, if you want to change this completely there is no other option available than overriding the entire template in child-theme for now. We can remove the code here and use an action hook instead, then you can remove the action and add your own one. But is it worth?

> And if aria-label is already added to the element, then the nested span.visually-hidden-focusable is no longer necessary because aria-label already does the job.

Any improvements are welcome. Please open a pull request, then we can merge your changes direct into the theme.

cc/ @massimoghm

**Comment by massimoghm** [comment]
@neralex thanks.

Which test tool do you use?

Regards

**Comment by neralex** [comment]
@massimoghm I'm familar with the WCAG rules since some years based on my work in the last 6 years and those simple rules are much better than all the tools, which you can find on the internet because accessibility is not measurable, you have to balance it always with the human factor. 

Tools can give you only a small point of view along the technical way but not about what makes sense or not for humans with different disabilities. You can test a lot with some tools and you can try to fix it all but you will find everytime something, which you can simplify a bit more for everyone.

For checking colour contrasts I'm using the Colour Contrast Analyser (CCA), which is available for Windows and macOS and also as a browser-plugin for different browsers but I really like the native app. I'm using NVDA with the eSpeak engine to test websites with a screenreader because this app is just simple to use and you can built a transportable version, which you can just copy on a usb-stick and take it with you. For checking headlines I'm using the HeadingsMap browser-plugin and the WAVE tool for verfiying overlooked issues. But the end you have to verify it by your own, which changes are making sense or not.

#keepitsimple #keepitclean
